src/mrsimulator/method/frequency_contrib.py

```diff
@@ -85,62 +150,97 @@ class FrequencyEnum(str, Enum):
+    ``"First_order"``:
+        Selects all first-order interactions
+    ``"Second_order"``:
+        Selects all second-order interactions
+        Selects
+    ``"Zeroth_rank"``:
+        Selects all zeroth-rank interactions
+    ``"Second_rank"``:
+        Selects all second-rank interactions
+    ``"Fourth_rank"``:
+        Selects all fourth-rank interactions
     """
 
     # Shielding1: str = "Shielding1"
-    Shielding1_0: str = freq_list_all[0]
-    Shielding1_2: str = freq_list_all[1]
+    Shielding1_0: str = FREQ_LIST_ALL[0]
+    Shielding1_2: str = FREQ_LIST_ALL[1]
+
+    # All Shielding
+    # Shielding: set = {FREQ_LIST_ALL[0], FREQ_LIST_ALL[1]}
 
     # Quad1: str = "Quad1"
-    Quad1_2: str = freq_list_all[2]
+    Quad1_2: str = FREQ_LIST_ALL[2]
 
     # Quad2: str = "Quad2"
-    Quad2_0: str = freq_list_all[3]
-    Quad2_2: str = freq_list_all[4]
-    Quad2_4: str = freq_list_all[5]
+    Quad2_0: str = FREQ_LIST_ALL[3]
+    Quad2_2: str = FREQ_LIST_ALL[4]
+    Quad2_4: str = FREQ_LIST_ALL[5]
 
     # J1: str = "J1"
-    J1_0: str = freq_list_all[6]
-    J1_2: str = freq_list_all[7]
+    J1_0: str = FREQ_LIST_ALL[6]
+    J1_2: str = FREQ_LIST_ALL[7]
 
     # D1: str = "D1"
-    D1_2: str = freq_list_all[8]
+    D1_2: str = FREQ_LIST_ALL[8]
 
     # Quad_Shielding: str = "Quad_Shielding"
-    Quad_Shielding_cross_0: str = freq_list_all[9]
-    Quad_Shielding_cross_2: str = freq_list_all[10]
-    Quad_Shielding_cross_4: str = freq_list_all[11]
+    Quad_Shielding_cross_0: str = FREQ_LIST_ALL[9]
+    Quad_Shielding_cross_2: str = FREQ_LIST_ALL[10]
+    Quad_Shielding_cross_4: str = FREQ_LIST_ALL[11]
 
-    Quad_J_cross_0: str = freq_list_all[12]
-    Quad_J_cross_2: str = freq_list_all[13]
-    Quad_J_cross_4: str = freq_list_all[14]
+    Quad_J_cross_0: str = FREQ_LIST_ALL[12]
+    Quad_J_cross_2: str = FREQ_LIST_ALL[13]
+    Quad_J_cross_4: str = FREQ_LIST_ALL[14]
 
-    Quad_Dipolar_cross_0: str = freq_list_all[15]
-    Quad_Dipolar_cross_2: str = freq_list_all[16]
-    Quad_Dipolar_cross_4: str = freq_list_all[17]
+    Quad_Dipolar_cross_0: str = FREQ_LIST_ALL[15]
+    Quad_Dipolar_cross_2: str = FREQ_LIST_ALL[16]
+    Quad_Dipolar_cross_4: str = FREQ_LIST_ALL[17]
 
     class Config:
         extra = "forbid"
 
-    # @validator("Quad1", pre=True, always=True)
-    # def validate_Quad1(cls, v, *, values, **kwargs):
-    #     values["Quad1_2"] = "Quad1_2"
-    #     return None
-
-    # @validator("Quad2", pre=True, always=True)
-    # def validate_Quad2(cls, v, *, values, **kwargs):
-    #     for item in ["Quad2_0", "Quad2_2", "Quad2_4"]:
-    #         values[item] = item
-    #     return None
-
     def json(self, **kwargs) -> str:
         """Parse the class object to a JSON compliant python dictionary object."""
         return self.value
 
+    # def dict(self, **kwargs) -> str:
+    #     """Parse the class object to a JSON compliant python dictionary object."""
+    #     return self.json()
```

Comment:
remove this

Reply:
Disregard, seralization function moved into Event class